pkg/component/worker/kubelet.go

```diff
@@ -131,6 +131,27 @@ func (k *Kubelet) Init(_ context.Context) error {
 	return nil
 }
 
+func getIPs(hostname string) (net.IP, net.IP, error) {
+	ips, err := net.LookupIP(hostname)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ipv4, ipv6 net.IP
+	for _, ip := range ips {
```

Comment:
Maybe pass on the context here?

```suggestion
func getIPs(ctx context.Context, hostname string) (net.IP, net.IP, error) {
	ips, err := net.DefaultResolver.LookupIPAddr(ctx, hostname)
	if err != nil {
		return nil, nil, err
	}

	var ipv4, ipv6 net.IP
	for _, ip := range ips {
		ip := ip.IP
```

Reply:
Seems like a good idea. https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/net/lookup.go;l=205